### ISSUE TITLE
printf %()T construct not recognized

### ISSUE THREAD

**Issue by charles-dyfis-net** [koalaman/shellcheck#775]
#### For bugs
- Rule id SC2183
- Shellcheck version: Online
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
printf '%(%Y%m%d)T\n' -1

```

#### Here's what shellcheck currently says:

```
Line 2:
printf '%(%Y%m%d)T\n' -1
       ^-- SC2183: This format string has 4 variables, but is passed 1 arguments.
```

#### Here's what I wanted or expected to see:

No issues detected!

---

Passing 0 arguments for a %T usage is also reasonable (default is `now`).

**Comment by kurahaupo** [comment]
It depends on which version of Bash you're using.

The %(...)X format control was introduced relatively recently (in 4.2 IIRC).
For any older version of Bash the warning is correct, but in addition %( was
not a valid conversion.

**Comment by slodki** [comment]
%()T was added in [bash 4.2](http://git.savannah.gnu.org/cgit/bash.git/commit/builtins/printf.def?id=495aee441b75276e38c75694ccb455bb6463fdb9), 5 years ago.

% chars inside () cannot be count as required parameters (empty format string, %()T is valid too). [checkPrintfVar](https://github.com/koalaman/shellcheck/blob/08f7ff37c590f8aa6dba6d989d707aad4adc5f4c/ShellCheck/Checks/Commands.hs#L473) should be rewritten to skip them.

If %()T is last format variable the corresponding printf parameter could be skipped (default will be used), so shellcheck shouldn't raise en error in this case.

**Comment by koalaman** [comment]
Nice. This is fixed in 838f0ce, though it still requires you to have a parameter for `%(..)T`
